src/utilstrencodings.cpp

```diff
@@ -544,3 +548,13 @@ bool ParseFixedPoint(const std::string &val, int decimals, int64_t *amount_out)
     return true;
 }
 
+#ifdef WIN32
+std::wstring Utf8ToWide(const std::string& utf8_string)
+{
+    size_t size = MultiByteToWideChar(CP_UTF8, 0, &*utf8_string.begin(), utf8_string.size(), nullptr, 0);
```

Comment:
I find the &* unnecessarily complicated. Can you use utf8_string.data() or even the more usual c_str() instead that gives a direct pointer back.

Reply:
This avoid the unnecessary copy, `&*utf8_string.begin()` point to where the exact string data are. `c_str()` and `data()` is a copy of the string data.

Reply:
Not sure what implementation you are referring to but c_str and data have always returned the pointer to the internal data. Not a copy. Anyway it was just a nitpick. Thanks.